tokio/src/park/thread.rs

```diff
@@ -21,8 +21,8 @@ pub(crate) struct UnparkThread {
 #[derive(Debug)]
 struct Inner {
     state: AtomicUsize,
-    mutex: Mutex<()>,
-    condvar: Condvar,
+    mutex: Arc<Mutex<()>>,
```

Comment:
So this is a bit unfortunate, but we need this because the owned version of the condvar does not allow us to externally wake up this parker (without being a future being polled from it).